src/field/field3d.cxx

```diff
@@ -72,21 +74,18 @@ Field3D::Field3D(const Field3D& f) : Field(f.fieldmesh), data(f.data) {
   fieldCoordinates = f.fieldCoordinates;
 }
 
-Field3D::Field3D(const Field2D& f) : Field(f.getMesh()) {
+Field3D::Field3D(const Field2D& f) : Field(f) {
 
   TRACE("Field3D: Copy constructor from Field2D");
 
   nx = fieldmesh->LocalNx;
   ny = fieldmesh->LocalNy;
   nz = fieldmesh->LocalNz;
 
-  location = f.getLocation();
-  fieldCoordinates = nullptr;
-
   *this = f;
 }
 
-Field3D::Field3D(const BoutReal val, Mesh* localmesh) : Field(localmesh) {
+Field3D::Field3D(const BoutReal val, Mesh* localmesh) : Field3D(localmesh) {
```

Comment:
This now delegates to the `Field3D` ctor that sets `nx` etc. so we can probably remove L92-94 below that also set them